Project/systick_interface.h

```
@@ -0,0 +1,97 @@
+/*
+**@file     : systick_interface.h
+**@brief    : this is header file to interface with Systick module 
+**@author   : sharpel
+**@date     : 19 june 2023
+**@version  : 0.1
+*/
+
+
+#ifndef SYSTICK_INTERFACE_H_
+#define SYSTICK_INTERFACE_H_
+
+/*includes*/
+
+#include "common.h"
+
+/*typdefs*/
+typedef void (*ptrf_systick_cb_t)(void);
+
+// error status
+typedef enum
+{
+	SYSTICK_OKAY           = 0,
+	INVALID_ARGUMENT_TIME,
+	NULL_CB_REF
+}enu_systick_status_t;	
+
+
+
+
+
+/*
+*@name       : SYSTICK_init
+*@berif      : this function init Systick Module
+*@param[in]  : void
+						
+					
+*@return     : void
+*/
+void SYSTICK_init (void);
+
+/*
+*@name       : SYSTICK_enable_interrupt
+*@berif      : this function enable interrupt in Systick Module
+*@param[in]  : void
+						
+					
+*@return     : void
+*/
+void SYSTICK_enable_interrupt (void);
```

every API should return an error state

---

Done, you can check it now

---

for your next task, I hope to  
1- make a main.c
2- make file structure

---

Okay dear, Thank you for reviewing

---

You are welcome